src/Domain.LinnApps/PurchaseOrders/PurchaseOrderService.cs

```diff
@@ -183,55 +183,162 @@ private void UpdateDetails(ICollection<PurchaseOrderDetail> currentDetails, ICol
+
+            this.UpdateOrderPostingsForDetail(current, updated);
+
+            this.UpdateDeliveries(current.PurchaseDeliveries, updated.PurchaseDeliveries);
+        }
+
+        private void UpdateDeliveries(ICollection<PurchaseOrderDelivery> deliveries, ICollection<PurchaseOrderDelivery> updatedDeliveries)
+        {
+            foreach (var delivery in deliveries)
+            {
+                var updatedDelivery = updatedDeliveries.First(x => x.DeliverySeq == delivery.DeliverySeq);
+                delivery.DateRequested = updatedDelivery.DateRequested;
+                //price updates to be done next
+                //delivery.OurUnitPriceCurrency = updatedDelivery.OurUnitPriceCurrency;
+                //delivery.OrderUnitPriceCurrency = updatedDelivery.OrderUnitPriceCurrency;
+                //delivery.BaseOrderUnitPrice = updatedDelivery.BaseOrderUnitPrice;
+                //delivery.BaseOurUnitPrice = updatedDelivery.BaseOurUnitPrice;
+                //delivery.VatTotalCurrency = updatedDelivery.VatTotalCurrency;
+                //delivery.BaseVatTotal = updatedDelivery.BaseVatTotal; //// vat totals might not change
+                //delivery.DeliveryTotalCurrency = updatedDelivery.DeliveryTotalCurrency;
+                //delivery.BaseDetailTotal = updatedDelivery.BaseDetailTotal;
+            }
+        }
+
+            private void UpdateMiniOrder(PurchaseOrder updatedOrder)
+        {
+            var miniOrder = this.miniOrderRepository.FindById(updatedOrder.OrderNumber);
+            var updatedDetail = updatedOrder.Details.First();
+
+            miniOrder.OurQty = updatedDetail.OurQty;
+            miniOrder.OrderQty = updatedDetail.OrderQty;
+
+            miniOrder.OurPrice = updatedDetail.OurUnitPriceCurrency;
+            miniOrder.OrderPrice = updatedDetail.OrderUnitPriceCurrency;
+            miniOrder.NetTotal = updatedDetail.NetTotalCurrency;
+            miniOrder.BaseOurPrice = updatedDetail.BaseOurUnitPrice;
+            miniOrder.BaseOrderPrice = updatedDetail.BaseOrderUnitPrice;
+            miniOrder.BaseNetTotal = updatedDetail.BaseNetTotal;
+
+            miniOrder.Remarks = updatedOrder.Remarks;
+            miniOrder.Department = updatedDetail.OrderPosting.NominalAccount.Department.DepartmentCode;
+            miniOrder.Nominal = updatedDetail.OrderPosting.NominalAccount.Nominal.NominalCode;
+            miniOrder.RequestedDeliveryDate = updatedDetail.PurchaseDeliveries.First().DateRequested;
+            miniOrder.InternalComments = updatedDetail.InternalComments;
+            miniOrder.SuppliersDesignation = updatedDetail.SuppliersDesignation;
+        }
+
+        /// <summary>
+        /// /below method currently unreferenced, but leaving
+        /// as it has most of what we'll need to put into mini order when creating
+        /// so will save re-doing it when we build create
+        /// </summary>
+        /// <param name="order"></param>
```

Comment:
I would just stick this in a notepad or something as opposed to merging unreferenced code